pico/api/src/main/java/io/helidon/pico/api/Helidon.java

```diff
@@ -0,0 +1,133 @@
+    private static final ReentrantReadWriteLock.ReadLock READ_LOCK = REENTRANT_READ_WRITE_LOCK.readLock();
+    private static final ReentrantReadWriteLock.WriteLock WRITE_LOCK = REENTRANT_READ_WRITE_LOCK.writeLock();
+    private static final AtomicBoolean BASIC_INIT_DONE = new AtomicBoolean();
+    private static final AtomicBoolean REGISTRY_INITIALIZED = new AtomicBoolean();
+    private static final AtomicBoolean STARTED = new AtomicBoolean();
+
+    static {
+        LogConfig.initClass();
+        ResettableHandler.registerReset();
+    }
+
+    private Helidon() {
+    }
+
+    /**
+     * Initialize Helidon Injection service registry. In case the intent is to also start services, such as WebServer,
+     * see {@link #start()}.
+     *
+     * @return service registry
+     */
+    public static Services serviceRegistry() {
+        if (REGISTRY_INITIALIZED.compareAndSet(false, true)) {
+            basicInit();
+            registryInit(false);
+        }
+        try {
+            READ_LOCK.lock();
+            return PicoServices.realizedServices();
+        } finally {
+            READ_LOCK.unlock();
+        }
+    }
+
+    /**
+     * Initialize Helidon Injection service registry, and start all startable services.
+     */
+    public static void start() {
+        if (STARTED.compareAndSet(false, true)) {
+            basicInit();
+            registryInit(true);
+        } else {
+            LOGGER.log(System.Logger.Level.WARNING, "Duplicate call to method Helidon.start().");
+        }
+    }
+
+    private static boolean reset(boolean deep) {
+        // for testing purposes
+        BASIC_INIT_DONE.set(false);
+        REGISTRY_INITIALIZED.set(false);
+        STARTED.set(false);
+
+        return true;
+    }
+
+    private static void registryInit(boolean bootServices) {
+        try {
+            WRITE_LOCK.lock();
+
+            boolean explicitConfig = GlobalConfig.configured();
+            Config bootstrapConfig = GlobalConfig.config();
+
+            Bootstrap bootstrap = Bootstrap.builder()
+                    .config(bootstrapConfig)
+                    .build();
+
+            PicoServices.globalBootstrap(bootstrap);
+            Services services = PicoServices.realizedServices();
+
+            if (!explicitConfig) {
+                GlobalConfig.config(() -> services.lookup(Config.class).get(), true);
+            }
+
+            if (bootServices) {
+                services.lookupAll(Startable.class)
+                        .stream()
+                        .map(ServiceProvider::get)
+                        .forEach(Startable::startService);
+            }
+        } finally {
+            WRITE_LOCK.unlock();
+        }
+    }
+
+    private static void basicInit() {
+        if (BASIC_INIT_DONE.compareAndSet(false, true)) {
+            LogConfig.configureRuntime();
+        }
+    }
+
+    private static final class ResettableHandler extends PicoServicesHolder {
+        @Deprecated
```

Comment:
Why `@Deprecated`, `ResettableHandler` is declared private.

Reply:
because the super constructor is deprecated, and this will disable a warning